source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -78,21 +78,25 @@ namespace
         { "u_time", "Time" }
     };
 
+    namespace OgsInputFlags
+    {
+        static const std::string
+            VARYING_INPUT_PARAM = "varyingInputParam",
+            IS_REQUIREMENT_ONLY = "isRequirementOnly";
+    }
+
+
     // Custom flags required by OGS XML
     static const StringMap OGS_FLAGS_MAP = 
     {
-        // Note: Nw has to be a varyingInputParam instead of a global (isRequirementOnly).
-        // This is because the pixel shader main function generated preprocesses Nw before passing it 
-        // to the main function of the fragment graph.
-        //
-        { "Pw", "isRequirementOnly" },
-        { "Pm", "isRequirementOnly" },
-        { "Nw", "varyingInputParam" },
-        { "Nm", "isRequirementOnly" },
-        { "Tw", "isRequirementOnly" },
-        { "Tm", "isRequirementOnly" },
-        { "Bw", "isRequirementOnly" },
-        { "Bm", "isRequirementOnly" } 
+        { "Pw", OgsInputFlags::VARYING_INPUT_PARAM },
```

Comment:
Pw and Tw are now also routed through the fragment root function's arguments to use the same vertex data mechanism between GLSL and HLSL.